core/src/smartcontracts/isi/triggers/set.rs

```diff
@@ -0,0 +1,644 @@
+            // Ignoring `None` variant because this means that action was deleted after `handle_*()`
+            // call and before `inspect_matching()` call
+            let result = match event {
+                Event::Data(_) => self
+                    .data_triggers
+                    .get(&id)
+                    .map(|entry| apply_f(entry.key(), entry.value(), event)),
+                Event::Pipeline(_) => self
+                    .pipeline_triggers
+                    .get(&id)
+                    .map(|entry| apply_f(entry.key(), entry.value(), event)),
+                Event::Time(_) => self
+                    .time_triggers
+                    .get(&id)
+                    .map(|entry| apply_f(entry.key(), entry.value(), event)),
+                Event::ExecuteTrigger(_) => self
+                    .by_call_triggers
+                    .get(&id)
+                    .map(|entry| apply_f(entry.key(), entry.value(), event)),
+            };
+
+            match result.flatten() {
+                Some(Ok(_)) => succeed.push(id),
+                Some(Err(err)) => errors.push(err),
+                None => {}
+            };
+        }
+
+        if errors.is_empty() {
+            return (succeed, Ok(()));
+        }
+        (succeed, Err(errors))
+    }
+
+    /// Remove actions with zero execution count from `triggers`
+    fn remove_zeros<F: Filter>(&self, triggers: &DashMap<TriggerId, LoadedAction<F>>) {
+        let to_remove: Vec<TriggerId> = triggers
+            .iter()
+            .filter_map(|entry| {
+                if let Repeats::Exactly(atomic) = &entry.value().repeats {
+                    if atomic.get() == 0 {
+                        return Some(entry.key().clone());
+                    }
+                }
+                None
+            })
+            .collect();
+
+        for id in to_remove {
+            triggers
+                .remove(&id)
+                .and_then(|_| self.ids.remove(&id))
+                .expect(
+                    "Removing existing keys from `Set` should be always possible. This is a bug",
+                );
+        }
+    }
+}
+
+#[derive(Clone)]
+pub enum LoadedExecutable {
+    Wasm(wasmtime::Module),
+    Instructions(Vec<InstructionBox>),
+}
+
+impl core::fmt::Debug for LoadedExecutable {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Wasm(_) => f
+                .debug_tuple("Wasm")
+                .field(&"<Wasm module is truncated>")
+                .finish(),
+            Self::Instructions(instructions) => {
+                f.debug_tuple("Instructions").field(instructions).finish()
+            }
+        }
+    }
+}
+
+impl From<LoadedExecutable> for Executable {
+    fn from(executable: LoadedExecutable) -> Self {
+        match executable {
+            LoadedExecutable::Wasm(module) => Executable::Wasm(WasmSmartContract::new(
+                module
+                    .serialize()
```

Comment:
We should inform users that obtained bytes are not wasm smartcontract, but wasmtime inner representation of module.

Reply:
I don't know the best way to do it. Should I introduce a new enum just for that?

Reply:
>  Should I introduce a new enum just for that?

Not sure either. What do you think about returning `Trigger<FilterBox, LoadedExecutable>` we can define `Encode` and `Decode` for `LoadedExecutable`?

Reply:
I'm afraid it won't work because we have to specify query return type in `iroha_data_model`, but I think we don't want `wasmtime` to be a dependency of `iroha_data_model`

Reply:
Oh, i see, so we don't have that much options. So i think you should add another enum.

Reply:
Check out new version, please